### ISSUE TITLE
UploadFromClasspath fails when the file is inside a jar

### ISSUE THREAD

**Issue by torrlane** [selenide/selenide#1493]
## The problem
I'm running my tests as maven integration-tests using the maven-failsafe plugin. As part of these tests I would like to upload some files to the app - however, those files are under src/main/resources and so are bundled into the application jar. I think that this is causing the following error:

```java
java.lang.IllegalArgumentException: URI is not hierarchical
        at java.base/java.io.File.<init>(File.java:420)
        at com.codeborne.selenide.commands.UploadFileFromClasspath.findFileInClasspath(UploadFileFromClasspath.java:44)
        at com.codeborne.selenide.commands.UploadFileFromClasspath.execute(UploadFileFromClasspath.java:30)
        at com.codeborne.selenide.commands.UploadFileFromClasspath.execute(UploadFileFromClasspath.java:19)
        at com.codeborne.selenide.commands.Commands.execute(Commands.java:155)
        at com.codeborne.selenide.impl.SelenideElementProxy.dispatchAndRetry(SelenideElementProxy.java:133)
        at com.codeborne.selenide.impl.SelenideElementProxy.invoke(SelenideElementProxy.java:85)
        at com.sun.proxy.$Proxy455.uploadFromClasspath(Unknown Source)
```

It looks like the `uploadFromClasspath` function does not support resources that are inside jars. Any help would be very much appreciated. Also, thank you for Selenide - it's possibly the best library that I've ever worked with.

**Comment by asolntsev** [comment]
@torrlane Why do you need to use a file from jar? Don’t you have tests in `src/test/java` and resources (including files to be uploaded) in `src/test/resources`?

**Comment by torrlane** [comment]
It's complicated (probably without good reason). I have an 'integration-test' submodule . The src/main/java contains all of the code for setting up the environment and the code for manipulating the UI. the src/test/ folder contains just the tests (which call the code in src/main/java). there are lots of cases where I need to upload files without really caring about their contents (just that they are different from other files that might have been uploaded). To save on typos, I used an enum to allow autocompletion and for my code to type check that a 'file' object was being passed in. Since that was now part of the 'api' for the integration tests, it had to go in src/main/java.

However, writing this out and explaining it, I'm now beginning to think that this isn't the best way of doing things.  I'm going to change the code so that the files are in src/test/resources. Thankyou for your help :)

**Comment by torrlane** [comment]
Once I've made my changes, I shouldn't be affected by this anymore. I'm happy to close this issue if you want.

**Comment by asolntsev** [comment]
It seems it's a good idea to simplify your test resources structure. :)

But anyway, Selenide could upload any files from classpath, including JAR files. Probably we could still implement this feature.

**Comment by asolntsev** [comment]
@torrlane I implemented is so that Selenide first extracts the file from jar to a temporary folder, and then uploads that temporary file. Will be released in Selenide 5.23.0
